### PR TITLE
Update persistence dependencies

### PR DESCRIPTION
**Pull request by mabrcosta**
Upgrades:

- Hibernate: 4.2.21.Final -> 5.4.20.Final
- org.glassfish.javax.el: 2.2.6 -> 3.0.0
- Querydsl: 4.2.21.Final -> 4.3.1

### CODE REVIEW

File: billy-portugal/src/main/java/com/premiumminds/billy/portugal/persistence/dao/jpa/AbstractDAOPTGenericInvoiceImpl.java [hunk 1]
```diff
@@ -44,10 +44,13 @@ public TInterface findByNumber(UID uidBusiness, String number) {
...
-                                .eq(uidBusiness.toString()).and(invoice.number.eq(number)))
-                        .singleResult(invoice),
-                PTGenericInvoiceEntity.class);
+            this.createQuery()
+                .from(invoice)
+                .where(this.toDSL(invoice.business, QJPAPTBusinessEntity.class).uid
+                           .eq(uidBusiness.toString())
+                           .and(invoice.number.eq(number)))
+                .select(invoice)
+                .fetchOne(),
```
Comment (rlapao) [inline billy-portugal/src/main/java/com/premiumminds/billy/portugal/persistence/dao/jpa/AbstractDAOPTGenericInvoiceImpl.java:53]: .fetchFirst() ?

**Review by rlapao — APPROVED** [review]
Não estou certo daquela alteração que fiz o comentário, de resto não encontrei mais nada.

**Review by rlapao — APPROVED** [review]
Ok.

**Review by rlapao — APPROVED** [review]
Ok.

File: billy-portugal/src/main/java/com/premiumminds/billy/portugal/persistence/dao/jpa/AbstractDAOPTGenericInvoiceImpl.java (hunk 1 above)
Comment (mabrcosta) [reply billy-portugal/src/main/java/com/premiumminds/billy/portugal/persistence/dao/jpa/AbstractDAOPTGenericInvoiceImpl.java:53]: As opposed to what we discussed earlier on the project with the library usage, here it was decided to also fix the behavior. 
Just the migration would be `fetchFirst()`, as you sugested.

**Review by froque — APPROVED** [review]

**Pull request merged**